texastribune/queso-ui
# Add a task for creating date labels

#### What's this PR do?

Adds an `npm run dates` task, which uses [node-canvas](https://github.com/Automattic/node-canvas) to dynamically create branded, published x year images. 

##### Assets added
- images/ - This is a new directory that will be shared along with other npm assets like scss and svg files. In the future, we could also store logo assets here.

#### Why are we doing this? How does it help us?

Ultimately, the goal is similar to what the Guardian wrote about [here](https://www.niemanlab.org/2019/04/the-guardians-nifty-old-article-trick-is-a-reminder-of-how-news-organizations-can-use-metadata-to-limit-misinformation/).

We'll house these images in this repo to keep them out of the way in our main repo. Having them also helps them stay versioned along with other design system-related changes. A design change to the labels would merit a new npm release.


#### How should this be manually tested?
`npm install`
`npm run dates`

This should print `✔ Generating date labels` in the terminal and build 1999 - 2021 date labels.


#### Does this introduce a breaking change where queso-ui is used in the wild? If so, is there a relevant branch/PR to accompany this release?

After I pre-release this, I'll add a simple script in our main repo to copy these images into our static_root folder. We'll then use thumbor to reference them as overlays.
